### PR TITLE
GH-15001: [R] Fix Parquet datatype test failure

### LINKED ISSUE #15001: [R] Map/list equality tests are failing only on the sanitizer nightlies
### Describe the bug, including details regarding any error messages, version, and platform.

I don't know why this is happening only on the sanitizer nightlies ([test-ubuntu-18.04-r-sanitizer](https://github.com/ursacomputing/crossbow/runs/10106173403) and [test-fedora-r-clang-sanitizer](https://github.com/ursacomputing/crossbow/runs/10130820766)) , but it seems to be the case.

These tests were added in #14847. @wjones127 any ideas?

### Component(s)

R

### PR DESCRIPTION
**Pull request by wjones127**
* Closes: #15001

### CODE REVIEW

**Comment by wjones127** [comment]
@github-actions crossbow submit -g r

File: r/tests/testthat/test-parquet.R [hunk 1]
```diff
@@ -453,12 +453,28 @@ test_that("deprecated int96 timestamp unit can be specified when reading Parquet
 })
 
 test_that("Can read parquet with nested lists and maps", {
-  parquet_test_data <- test_path("../../../cpp/submodules/parquet-testing/data")
+  test_that("Can read parquet with nested lists and maps", {
+  # Construct the path to the parquet-testing submodule. This will search:
```
Comment (wjones127) [inline r/tests/testthat/test-parquet.R:457]:
```suggestion
  # Construct the path to the parquet-testing submodule. This will search:
```

**Review by jonkeane — APPROVED** [review]
Thanks for these test fixups!

**Comment by wjones127** [comment]
@jonkeane I think the test was still skipped

https://github.com/apache/arrow/actions/runs/3847573332/jobs/6554198668#step:9:9689

Did you see it run in one of the CI jobs?

**Comment by jonkeane** [comment]
Ack, you're right. I had seen it run, but it must have been a prior run and not that latest one. I'll make an issue + send a PR I think I know what the issue is.

**Pull request merged**